central/pruning/pruning.go

```diff
@@ -461,6 +469,26 @@ func (g *garbageCollectorImpl) removeOrphanedPLOP() {
 	}
 }
 
+func (g *garbageCollectorImpl) removeExpiredAdministrationEvents(config *storage.PrivateConfig) {
+	retentionDays := config.GetAdministrationEventsConfig().GetRetentionDurationDays()
+	retentionQuery := search.NewQueryBuilder().AddDays(search.LastUpdatedTime, int64(retentionDays)).ProtoQuery()
+	results, err := g.administrationEvents.Search(pruningCtx, retentionQuery)
```

Comment:
We have been trying to move away from this pattern where we walk all the data and then remove by IDs.  The pattern here of searching for everything in a time window is essentially the same as it it is a 2 step process when it could be a 1 step process.  See https://github.com/stackrox/stackrox/blob/master/central/postgres/pruning.go.

Obviously outside scope, but we (probably me) may want to investigate generating pruning functions for the `delete all records older than` pattern.

Reply:
Ah gotcha, changed it to follow the new pattern.

If we were to generate those functions, probably would also be an idea to move them to the generated store + have the datastore expose the function, this way we could possible introduce metrics with on datastore level to track execution times for those pruning functions (theoretically could do that as well right now, but would feel more natural to the patterns that currently exist).

Reply:
https://issues.redhat.com/browse/ROX-19730
https://issues.redhat.com/browse/ROX-19731

No clue when we will get to them.